### ISSUE TITLE
[BUG] Base class `BaseTwoIndexSymmetric` in base_two_symm.py can not deal with mix cartesian/spherical coord_types

### ISSUE THREAD

**Issue by leila-pujal** [theochem/gbasis#119]
**Describe the bug**
 
I was trying to compute kinetic energy atomic integrals with `gbasis.integrals.kinetic_energy.
kinetic_energy_integral()` with a fchk file where there are mix cartesian and spherical shells (
[neon.fchk.tar.gz](https://github.com/theochem/gbasis/files/6337979/neon.fchk.tar.gz) ) with the code in Reproduce section and got the next error: 

```
File "issue_code.py", line 11, in <module>
    kinetic_integrals = kinetic_energy_integral(basis, coord_type = coord)
  File "/home/leila/gbasis/gbasis/gbasis/integrals/kinetic_energy.py", line 159, in kinetic_energy_integral
    return KineticEnergyIntegral(basis).construct_array_mix(coord_type)
  File "/home/leila/gbasis/gbasis/gbasis/base_two_symm.py", line 304, in construct_array_mix
    cont_one.angmom_components_sph,
  File "/home/leila/gbasis/gbasis/gbasis/wrappers.py", line 117, in angmom_components_sph
    "momentum {0}".format(self.angmom)
ValueError: Given convention does not support spherical contractions for the angular momentum 0
```
If I more or less understood the code correctly in gbasis, this happens because when the code in kinetic_energy.py tries to construct the contractions array with `KineticEnergyIntegral(basis).construct_array_mix(coord_type)` (coord_type = coord in the reproduce code and coord being the list of cartesian/spherical values for each shell) it goes to the base class module in `base_two_symm.py`, to use `construct_array_mix()` function defined there and in this case it builds the transformation variable for all the basis functions https://github.com/theochem/gbasis/blob/c57de8f1cfbe1dbaac8941c7cb9993797c76fca6/gbasis/base_two_symm.py#L301-L326, and after that checks which ones are spherical and applies the transformation. This provokes that the code checks if there is a convention for spherical basis functions in the conventions stored in mol object from IOData but in conventions there is only defined cartesian conventions for angular momentums 0 and 1 because there are the same in both cartesian and spherical. 

In contrast to this behaviour for `BaseTwoIndexSymmetric`, for `BaseOneIndex` in base_one.py this  sorting is done before building the transform variable and avoiding this mistake. 
 https://github.com/theochem/gbasis/blob/c57de8f1cfbe1dbaac8941c7cb9993797c76fca6/gbasis/base_one.py#L229-L234 

**To Reproduce**

```
#! usr/bin/env python
from iodata import load_one

from gbasis.wrappers import from_iodata
from gbasis.integrals.kinetic_energy import kinetic_energy_integral, KineticEnergyIntegral
import numpy as np

mol = load_one("neon.fchk")
basis, coord = from_iodata(mol)

kinetic_integrals = kinetic_energy_integral(basis, coord_type = coord)
```

**Expected behaviour**

**Screenshots**

**System Information:**
 - OS: Ubuntu 18.04.5 LTS
 - Python version: 3.6.10
 - NumPy version: 1.18.1
 - SciPy version: 1.4.1

**Additional context**

**Comment by leila-pujal** [comment]
I think this is the same bug reported in #114

**Comment by leila-pujal** [comment]
This was fixed in PR #120
